tests/examples/CMakeLists.txt.in

```diff
@@ -0,0 +1,45 @@
+#
+# We now parse all diff files and create rules to generate the step-*.cc
+# files.
+#
+# Here, ${_base_directory} points to the base "tests/examples" directory,
+# and ${DEAL_II_SOURCE_DIR} points to the deal.II main repository root.
+#
+
+file(GLOB _diff_files "${_base_directory}/*.diff")
+foreach(_diff_file ${_diff_files})
+  get_filename_component(_step "${_diff_file}" NAME_WLE)
```

Comment:
Does this support more than one diff file for a single step? It would be neat if step-40.petsc.diff and step-40.trilinos.diff could be made to support variations. Can be a different PR of course.

Reply:
@tjhei Yes, that would be easily doable.

Reply:
@tjhei Let's handle this in a separate PR.